app/src/main/java/gateway/controller/CtrlFileList.java

```diff
@@ -0,0 +1,95 @@
+package gateway.controller;
+
+import gateway.config.Config;
+import gateway.services.ServiceAuth;
+import gateway.soap.request.ReqFileList;
+import gateway.soap.response.File;
+import gateway.soap.response.ResFileList;
+import gateway.soap.response.ResStatus;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.UUID;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class CtrlFileList
+{
+	public static ResFileList file_list (ReqFileList args)
+	{
+		// Create a new instance of ResFileList to store the results
+		ResFileList resFileList = new ResFileList ();
+		UUID userUUID;
+		String url;
+
+		// User authentication
+		ResStatus resAuth = ServiceAuth.authenticate (args.token);
+		if (resAuth.error) {
+			return ResStatus.downCast (ResFileList.class, resAuth);
+		}
+
+		// Get the user's UUID from the authentication token
+		userUUID = UUID.fromString (ServiceAuth.tokenGetClaim (args.token, "uuid"));
+
+		// Build the URL for the request.
+		if (args.location == null) {
+			// If no location is specified, build the URL without a parentUUID parameter
+			url = Config.getMetadataBaseUrl () + "/files/list/" + userUUID;
+		} else {
+			// If a location is specified, include the parentUUID parameter
+			url = Config.getMetadataBaseUrl () + "/files/list/" + userUUID +
+				  "?parentUUID=" + args.location;
+		}
+
+		try {
+			// Make an HTTP GET request
+			HttpResponse<String> response = HttpClient.newHttpClient ().send (
+				HttpRequest.newBuilder ().uri (URI.create (url)).GET ().build (),
+				HttpResponse.BodyHandlers.ofString ());
+
+			JSONObject responseBody = new JSONObject (response.body ());
+			resFileList.code = response.statusCode ();
+
+			if (resFileList.code == 200) {
+				// If the response code is 200, process the received files
+				JSONArray filesArray = responseBody.getJSONArray ("files");
+				File[] files = new File[filesArray.length ()];
+
+				for (int i = 0; i < filesArray.length (); i++) {
+					// Process each file in the response JSON
+					JSONObject fileObject = filesArray.getJSONObject (i);
+
+					// Create a File object and assign it the file information
+					File file = new File ();
+
+					file.uuid = UUID.fromString (fileObject.getString ("uuid"));
+					file.name = fileObject.getString ("name");
```

Comment:
Some fields were changed in the most recent release, for instance, `name` was changed to `fileName` according to the [metadata spec](https://github.com/hawks-atlanta/metadata-scala/blob/main/docs/spec.openapi.yaml). Sorry for the inconveniences.